### PR TITLE
issue: HPCINFRA-1321 Switch cppcheck to a docker

### PR DESCRIPTION
**Pull request by vialogi**
## Description
We're going to move static tests from a dedicated server to docker containers to reduce CI timings 

##### What
Switch Cppcheck, Csbuild and Tidy steps of the CI to run in docker containers in a parallel mode

##### Why ?
CI currently takes too long, a majority of the time is consumed by static tests running on a physical machine

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

**Comment by igor-ivanov** [comment]
it seems that suggested container does not have `cppcheck` tool installed.
```
[2024-03-05T14:27:11.989Z] + echo '[SKIP] cppcheck tool does not exist'
[2024-03-05T14:27:11.989Z] [SKIP] cppcheck tool does not exist
```

Container should include cppcheck, csbuild and tidy the same version as beni09 to avoid extra noises in results.

File: .ci/dockerfiles/Dockerfile.rhel8.6 [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ARG ARCH=x86_64
+FROM harbor.mellanox.com/hpcx/$ARCH/rhel8.6/core:latest
+ARG MOFED_VER=5.3-1.0.0.1
```
Comment (dpressle) [inline .ci/dockerfiles/Dockerfile.rhel8.6:3]: not used please remove

File: .ci/matrix_job.yaml [hunk 2]
```diff
@@ -32,8 +33,8 @@ volumes:
   # User profile for release
   - {mountPath: /var/home/swx-jenkins, hostPath: /labhome/swx-jenkins}
 
-env:
-  build_dockers: false
+# env:
```
Comment (dpressle) [inline .ci/matrix_job.yaml:36]: remove comments

File: contrib/jenkins_tests/cppcheck.sh [hunk 3]
```diff
@@ -8,19 +8,21 @@ tool_app=cppcheck
...
-            if [ $? -eq 0 ]; then
-                tool_app=$PWD/cppcheck
-            fi
-        fi
-        cd ..
-    fi
-    set -e
+    yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+    yum install -y cppcheck
+    # set +e
```
Comment (dpressle) [inline contrib/jenkins_tests/cppcheck.sh:13]: If not needed lets remove

**Review by dpressle — CHANGES_REQUESTED** [review]

File: contrib/jenkins_tests/cppcheck.sh [hunk 4]
```diff
@@ -8,19 +8,21 @@ tool_app=cppcheck
...
-        if [ $? -eq 0 ]; then
-            eval "make $make_opt FILESDIR=$PWD HAVE_RULES=yes " > /dev/null 2>&1
-            if [ $? -eq 0 ]; then
-                tool_app=$PWD/cppcheck
-            fi
-        fi
-        cd ..
-    fi
-    set -e
+    yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
```
Comment (dpressle) [inline contrib/jenkins_tests/cppcheck.sh:11]: why its still here? its in the image now

**Comment by vialogi** [comment]
bot:retest

**Review by dpressle — APPROVED** [review]

**Pull request merged**